courses/utils.py

```diff
@@ -122,3 +128,74 @@ def generate_program_certificate(user, program):
         )
 
     return program_cert, True
+
+
+def revoke_program_certificate(user, readable_id, revoked_state, force_revoked):
+    """
+    Revoked a program certificate.
+
+    Args:
+        user (User): a Django user.
+        readable_id: represents the program (readable_id) for revoking a ProgramCertificate.
+        revoked_state: (bool) override the is_revoked state of ProgramCertificate.
+        force_revoked: (bool) Indicate to revoke all course runs that are associated with a program.
+    """
+    try:
+        program = Program.objects.get(readable_id=readable_id)
+    except Program.DoesNotExist:
+        raise
+
+    try:
+        program_certificate = ProgramCertificate.objects.get(
+            user=user, program__readable_id=readable_id
+        )
+    except ProgramCertificate.DoesNotExist:
+        log.warning(
+            "Program certificate for user: %s in program %s does not exist.",
+            user.username,
+            readable_id,
+        )
+        return False
+
+    program_certificate.is_revoked = revoked_state
+    program_certificate.save()
+
+    if force_revoked:
+        courses_in_program_ids = set(program.courses.values_list("id", flat=True))
+        CourseRunCertificate.objects.filter(
+            user=user, course_run__course_id__in=courses_in_program_ids
+        ).update(is_revoked=revoked_state)
+
+    return True
+
+
+def revoke_course_run_certificate(user, courseware_id, revoked_state):
+    """
+        Revoked a course run certificate.
+
+        Args:
+            user (User): a Django user.
+            courseware_id: represents the course run.
+            revoked_state: represents the course run (courseware_id) for revoking a CourseRunCertificate.
+    """
+    try:
+        course_run = CourseRun.objects.get(courseware_id=courseware_id)
+    except CourseRun.DoesNotExist:
+        raise
```

Comment:
I don't see any side-effect of this `try:except` here

Reply:
I think, I forgot to add logs there.